enhancements/sig-policy/85-gatekeeper-policy-integration/README.md

```diff
@@ -0,0 +1,258 @@
+# Native Support for Gatekeeper Constraints in Policies
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+      [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+Enhance the Open Cluster Management (OCM) policy framework to allow Gatekeeper `ConstraintTemplate` and constraint
+manifests to be directly embedded in OCM `Policy` object's `policy-templates` array for multicluster distribution and
+status aggregation.
+
+## Motivation
+
+The Open Cluster Management (OCM) Policy addon does not have native support for an admission controller. Gatekeeper is
+currently favored by the OCM Policy SIG for its flexibility and performance. Though the installation of Gatekeeper will
+not be simplified by this enhancement, there is a desire to provide native OCM Policy support for Gatekeeper
+`ConstraintTemplate` and constraint objects. This will allow users to skip using `ConfigurationPolicy` objects that wrap
+Gatekeeper constraint manifests. This also allows the audit messages from Gatekeeper constraints to be aggregated in the
+policy status on the OCM hub, which is not possible by the wrapping approach.
```

Comment:
If I was managing ConstraintTemplates and constraints with ConfigurationPolicies (without this enhancement), what do the violations look like?

Based on https://github.com/open-cluster-management-io/policy-collection/blob/main/community/CM-Configuration-Management/policy-gatekeeper-container-image-latest.yaml (just the first constraint example I found), it looks like to find violations without this enhancement, it checks the constraint's status.totalViolations count. Which means that the only message would be something generic like `violation - ____ found but not as expected`.

I think the improved messages possible from this enhancement are a big advantage.

Reply:
That's right. The error would be something like `violation - k8srequiredlabels not found: [ns-must-have-gk] found but not as specified`. I'll add that to the doc.